### ISSUE TITLE
Ellipsis with `ParamSpec` does not mean the same thing as with `Callable`

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#2725]
**Describe the bug**
Given the following code:

```python
from __future__ import annotations

from typing import Callable, Protocol
from typing_extensions import Concatenate, ParamSpec, TypeAlias

P = ParamSpec('P')


def foo(a: int, b: str) -> None:
    ...


class Handler(Protocol[P]):
    def __call__(self, /, *args: P.args, **kwargs: P.kwargs) -> None:
        ...


class ConcatHandler(Protocol[P]):
    def __call__(self, a: int, /, *args: P.args, **kwargs: P.kwargs) -> None:
        ...


ConcatCallableHandler: TypeAlias = Callable[Concatenate[int, P], None]


handler_callable: Callable[..., None] = foo
concat_handler_callable: ConcatCallableHandler[...] = foo

handler: Handler[...] = foo
concat_handler: ConcatHandler[...] = foo
```

Three errors are generated:

* Line 27:
```
Expression of type "(a: int, b: str) -> None" cannot be assigned to declared type "ConcatCallableHandler[(*args: Any, **kwargs: Any)]"
    Type "(a: int, b: str) -> None" cannot be assigned to type "ConcatCallableHandler[(*args: Any, **kwargs: Any)]"
      Parameter "*args" has no corresponding parameter
      Parameter "**kwargs" has no corresponding parameter
```
* Line 29:
```
Expression of type "(a: int, b: str) -> None" cannot be assigned to declared type "ConcatCallableHandler[(*args: Any, **kwargs: Any)]"
    Type "(a: int, b: str) -> None" cannot be assigned to type "ConcatCallableHandler[(*args: Any, **kwargs: Any)]"
      Parameter "*args" has no corresponding parameter
      Parameter "**kwargs" has no corresponding parameter
```
* Line 30:
```
Expression of type "(a: int, b: str) -> None" cannot be assigned to declared type "ConcatHandler[(*args: Any, **kwargs: Any)]"
    Type "(a: int, b: str) -> None" cannot be assigned to type "(a: int, /, *args: Any, **kwargs: Any) -> None"
      Parameter "*args" has no corresponding parameter
      Parameter "**kwargs" has no corresponding parameter
```

Because the ellipsis in `Callable[..., None]` indicates "any signature", I think it's a bug that the ellipsis in a `ParamSpec` does not indicate "any signature".

**Expected behavior**
No errors are generated.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line

Which version?
A built version of 348e8b2

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.197, which I just published. It will also be included in the next release of pylance.
